### PR TITLE
Update exif.h

### LINKED ISSUE #2100: jxl::FindExifTagPosition - pointing to memory that is out of bounds (Segfault)
**Describe the bug**

For verification, used the synthetic examples, which, however, is in appearance a regular files and can be opened through image viewer. After run, binary file aborted with Segfault.

**To Reproduce**
```bash
/run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl/tools/cjxl sig_11_src_000122_poc --disable_output
```
Output:
```bash
JPEG XL encoder v0.9.0 aa99c8d8 [AVX2,SSE4,SSSE3,Unknown]
Note: Implicit-default for JPEG is lossless-transcoding. To silence this message, set --lossless_jpeg=(1|0).
Read JPEG image with 94120 bytes.
Encoding [Container | JPEG, lossless transcode, effort: 7 | JPEG reconstruction data], 
Segmentation fault (core dumped)
```

```bash
Program terminated with signal SIGSEGV, Segmentation fault.
#0  jxl::FindExifTagPosition (exif=..., tagname=274) at /run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl_gdb/lib/jxl/exif.h:44
44	  uint16_t nb_tags = (bigendian ? LoadBE16(t) : LoadLE16(t));
[Current thread is 1 (Thread 0x7fd2f485a240 (LWP 3344243))]
(gdb) backtrace 
#0  jxl::FindExifTagPosition (exif=..., tagname=274)
    at /run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl_gdb/lib/jxl/exif.h:44
#1  jxl::InterpretExif (exif=std::vector of length 94, capacity 94 = {...}, 
    orientation=orientation@entry=0x7ffe000b4ee0)
    at /run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl_gdb/lib/jxl/exif.h:68
#2  0x00007fd2f6401e6e in JxlEncoderAddJPEGFrame (frame_settings=0x55d68f8664f0, buffer=<optimized out>, 
    size=94120)
    at /run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl_gdb/lib/jxl/encode.cc:1562
#3  0x000055d68f829c05 in jxl::extras::EncodeImageJXL (params=..., ppf=..., jpeg_bytes=0x7ffe000b59c0, 
    compressed=0x7ffe000b5980)
    at /run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl_gdb/lib/extras/enc/jxl.cc:82
#4  0x000055d68f81999c in main (argc=<optimized out>, argv=<optimized out>)
    at /run/media/user/b05c56b9-304e-42f6-af2b-883e9de3e6de/target/libjxl/sub/libjxl_gdb/tools/cjxl_main.cc:1041
```

**Expected behavior**
No assertion failed.

**Screenshots**
before patch:
56_old
affter patch:
56_new

**Environment**
 - OS: ```tested at 5.15.78-2.el7.3.x86_64 (RED OS release MUROM 7.3.2 - Centos based system)``` ;
 - Compiler version: ```clang version 12.0.1 (RED OS 12.0.1-1.el7)``` ;
 - CPU type: ```x86_64``` ;
 - cjxl version string: ```v0.9.0 aa99c8d8```

**Additional context**
For solution this problem added additional error checking to ensure that the t pointer is pointing to a valid location within the exif buffer before trying to read the value of nb_tags.
```cpp
if(t + 2 >= exif.data() + exif.size()) return 0;
```
(also added in a patch)

link to a sample (github-url):
[sig_11_src_000122_poc](https://github.com/a-shvedov/fuzzing_investigation/blob/main/res/libjxl/2022_01_25/sig_11_src_000122_poc)

### PR DESCRIPTION
**Pull request by a-shvedov**
trying to fix #2100

### CODE REVIEW

File: lib/jxl/exif.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ inline size_t FindExifTagPosition(const std::vector<uint8_t>& exif,
   uint32_t offset = (bigendian ? LoadBE32(t) : LoadLE32(t));
   if (exif.size() < 12 + offset + 2 || offset < 8) return 0;
   t += offset - 4;
+  if(t + 2 >= exif.data() + exif.size()) return 0; // gdb_affter_SIGSEGV: 44 uint16_t nb_tags = (bigendian ? LoadBE16(t) : LoadLE16(t));
```
Comment (mo271) [inline lib/jxl/exif.h:44]:
```suggestion
  uint64_t offset = (bigendian ? LoadBE32(t) : LoadLE32(t));
  if (exif.size() < 12 + offset + 2 || offset < 8) return 0;
  t += offset - 4;
  if (offset + 2 >= exif.size()) return 0; // gdb_affter_SIGSEGV: 44 uint16_t nb_tags = (bigendian ? LoadBE16(t) : LoadLE16(t));
```

**Comment by mo271** [comment]
Thanks for the suggested change, @a-shvedov! Could you follow the instructions above from the google-cla bot?!

**Comment by a-shvedov** [comment]
> Thanks for the suggested change, @a-shvedov! Could you follow the instructions above from the google-cla bot?!

Yes, sure!

File: lib/jxl/exif.h [hunk 2]
```diff
@@ -38,9 +38,10 @@ inline size_t FindExifTagPosition(const std::vector<uint8_t>& exif,
   bool bigendian;
   if (!IsExif(exif, &bigendian)) return 0;
   const uint8_t* t = exif.data() + 4;
-  uint32_t offset = (bigendian ? LoadBE32(t) : LoadLE32(t));
+  uint64_t offset = (bigendian ? LoadBE32(t) : LoadLE32(t));
   if (exif.size() < 12 + offset + 2 || offset < 8) return 0;
   t += offset - 4;
+  if (offset + 2 >= exif.size()) return 0; // gdb_affter_SIGSEGV: 44 uint16_t nb_tags = (bigendian ? LoadBE16(t) : LoadLE16(t));
```
Comment (veluca93) [inline lib/jxl/exif.h:44]: Can we remove the comment?

Comment (mo271) [reply lib/jxl/exif.h]: done

**Review by veluca93 — APPROVED** [review]

**Pull request merged**